web/src/components/storage/VolumeFields.test.jsx

```diff
@@ -0,0 +1,195 @@
+    // will not work because it's actually triggering the onChange with "/",
+    // "v", "a", and "r", but having a different "value" each time it's
+    // re-rendered. Anyway, checking that callback is called is enough.
+    await user.type(input, "/");
+    expect(callbackFn).toHaveBeenCalledWith("/");
+  });
+
+  it("renders only the value if mount as read-only (no input)", () => {
+    plainRender(<MountPathField value="/home" onChange={callbackFn} isReadOnly />);
+    expect(screen.queryByRole("textbox", { name: "Mount point" })).toBeNull();
+    screen.getByText("/home");
+  });
+});
+
+describe("SizeOptionsField", () => {
+  it("renders radio group with sizing options", () => {
+    plainRender(<SizeOptionsField formData={{ sizeMethod: "fixed" }} volume={volume} onChange={callbackFn} />);
+    screen.getByRole("radiogroup", { name: "Size" });
+  });
+
+  it("renders 'Fixed' and 'Range' options always but 'Auto' only if volume accepts auto size", () => {
+    const { rerender } = plainRender(
+      <SizeOptionsField formData={{ sizeMethod: "fixed" }} volume={volume} onChange={callbackFn} />
+    );
+    const group = screen.getByRole("radiogroup", { name: "Size" });
+    within(group).getByRole("radio", { name: "Fixed" });
+    within(group).getByRole("radio", { name: "Range" });
+    expect(within(group).queryByRole("radio", { name: "Auto" })).toBeNull();
+
+    rerender(
+      <SizeOptionsField
+        formData={{ sizeMethod: "fixed" }}
+        volume={{ ...volume, outline: { ...volume.outline, supportAutoSize: true } }}
+        onChange={callbackFn}
+      />
+    );
+    within(group).getByRole("radio", { name: "Auto" });
+    within(group).getByRole("radio", { name: "Fixed" });
+    within(group).getByRole("radio", { name: "Range" });
+  });
+
+  it("renders options as disabled when isDisabled props is given", () => {
+    plainRender(
+      <SizeOptionsField
+        formData={{ sizeMethod: "fixed" }}
+        volume={{ ...volume, outline: { ...volume.outline, supportAutoSize: true } }}
+        onChange={callbackFn}
+        isDisabled
+      />
+    );
+    const group = screen.getByRole("radiogroup", { name: "Size" });
+    within(group).getAllByRole("radio")
+      .forEach(option => expect(option).toBeDisabled());
+  });
+
+  it("calls given callback when user changes selected option", async () => {
+    const { user } = plainRender(
+      <SizeOptionsField formData={{ sizeMethod: "fixed" }} volume={volume} onChange={callbackFn} />
+    );
+    const group = screen.getByRole("radiogroup", { name: "Size" });
+    const rangeOption = within(group).getByRole("radio", { name: "Range" });
+    await user.click(rangeOption);
+    expect(callbackFn).toHaveBeenCalledWith({ sizeMethod: SIZE_METHODS.RANGE });
+  });
+
+  it.todo("test SizeAuto internal component");
+  it.todo("test SizeManual internal component");
+  it.todo("test SizeRange internal component");
```

Comment:
As said in a comment, left as "todo" because these components are going to change in a short term.